### PR TITLE
Document for the Edgecontroller module cloud part

### PR DESCRIPTION
**Pull request by srivatsav123**
This is the Readme.md for EdgeController

**What type of PR is this?**
 /kind documentation

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: docs/modules/cloudmodule/controller.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
+# Edge Controller
+
+
+## Edge Controller Overview
+ controller(also known as edgecontroller) is the bridge between EdgeMaster(Kubernetes Api Server) and Edgecore
```
Comment (sids-b) [inline docs/modules/cloudmodule/controller.md]: Kubernetes Api server is not edge master. Just keep it Kubernetes Api-Server

File: docs/modules/cloudmodule/controller.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+   
+## Operations Performed By Edge Controller
+ 
+ The following are the functions performed by Edge controller :-
+  - Register and Start Sub Modules
+  - Downstream Controller-Sync add/update/delete event to edgeCore from EdgeMaster
```
Comment (sids-b) [inline docs/modules/cloudmodule/controller.md]: Replace EdgeMaster to Api-Server everywhere please

Comment (sids-b) [reply docs/modules/cloudmodule/controller.md]: Follow this standard while writing explanation everywhere to keep consistency. Dont use '-'
```
FOO: BAR.....
```

File: docs/modules/cloudmodule/controller.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+
+## Edge Controller Overview
+ controller(also known as edgecontroller) is the bridge between EdgeMaster(Kubernetes Api Server) and Edgecore
+
+   
+## Operations Performed By Edge Controller
+ 
+ The following are the functions performed by Edge controller :-
+  - Register and Start Sub Modules
```
Comment (sids-b) [inline docs/modules/cloudmodule/controller.md]: This document is only for controller.... It should not include this

File: docs/modules/cloudmodule/controller.md [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  - Downstream Controller-Sync add/update/delete event to edgeCore from EdgeMaster
+  - Upstream Controller -sync watch and Update status of resource and events(node,pod and configmap) to Edge master and also subscribe message from edgecore
+  - Controller Manager-Creates manager Interface and implements ConfigmapManager LocationCache and podManager
+### Register and Start Sub Modules
```
Comment (sids-b) [inline docs/modules/cloudmodule/controller.md]: This is not done by controller, please remove. This document should only have information related to controller

File: docs/modules/cloudmodule/controller.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+- Registers two modules,controller and cloudhub and starts them as separate go routines 
+
+
+### Downstream Controller -Sync add/update/delete event to edge
+ 
+- Downstream controller-Watches Kubernetes Api server and send change to edge via cloudHub
```
Comment (sids-b) [inline docs/modules/cloudmodule/controller.md]: send change -> send updates

File: docs/modules/cloudmodule/controller.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+- Sync (pod,configmap,secret) add/update/delete event to edge via cloudHub
+- creates manager (pod,configmap,secret)for events by calling manager interface
```
Comment (sids-b) [inline docs/modules/cloudmodule/controller.md]: Not able to understand this line. Please give more clarification

File: docs/modules/cloudmodule/controller.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+- Locates configmap and secret should be send to which node
+
+![Downstream Controller](./../../images/edgecontroller/DownstreamController.png)
+### Upstream Controller -sync watch and Update status of resource and events
+
+- UpstreamController subscribes message from edge and sync to EdgeMaster(k8s api server)
```
Comment (sids-b) [inline docs/modules/cloudmodule/controller.md]: subscribes -> receives
message -> messages
sync -> syncs the updates

File: docs/modules/cloudmodule/controller.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+- Creates stop channel to dispatch,stopUpdatePodstatus,stopUpdateNodeStatus,stopQueryConfigMap and stopQuerySecret
```
Comment (sids-b) [inline docs/modules/cloudmodule/controller.md]: No need to give implementation details

File: docs/modules/cloudmodule/controller.md [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+- Creates message channel to update Nodestatus,Podstatus,Secret and configmap related events
+- Gets Podcondition information like Ready,initialized,Podscheduled and Unschedulable details
```
Comment (sids-b) [inline docs/modules/cloudmodule/controller.md]: Add a space after all ","

File: docs/modules/cloudmodule/controller.md [hunk 10]
```diff
@@ -0,0 +1,55 @@
...
+   - **Ready**-PodReady means the pod is able to service requests and should be added to the load balancing pools for all matching services
+   - **PodScheduled**-It represents status of the scheduling process for this pod
+   - **Unschedulable**-It means scheduler cant schedule the pod right now,may be due to insufficient resources in the cluster
```
Comment (sids-b) [inline docs/modules/cloudmodule/controller.md]: Have a sub-heading maybe to give an idea about what these headings are for ??

**Comment by pavan187** [comment]
> This is the Readme.md for EdgeController
> 
> **What type of PR is this?**
> documentation
> 
> **What this PR does / why we need it**:
> 
> **Which issue(s) this PR fixes**:
> 
> Fixes #
> 
> **Special notes for your reviewer**:

@srivatsav123 , Please follow PR description template, otherwise Ci-Bot Cannot assign Lables accordingly.

File: docs/modules/cloud/controller.md [hunk 11]
```diff
@@ -0,0 +1,54 @@
...
+
+## Edge Controller Overview
+ Controller(also known as edgecontroller) is the bridge between Kubernetes Api-Server and Edgecore
+
+   
+## Operations Performed By Edge Controller
+ 
+ The following are the functions performed by Edge controller :-
+  - Downstream Controller:Sync add/update/delete event to edgeCore from K8s Api-server
+  - Upstream Controller:sync watch and Update status of resource and events(node, pod and configmap) to K8s Api-server and also subscribe message from edgecore
```
Comment (anvithks) [inline docs/modules/cloud/controller.md]: Please use consistent naming conventions.
Edgecore or edgeCore or edgecore. (In the previous lines)

File: docs/modules/cloud/controller.md [hunk 12]
```diff
@@ -0,0 +1,54 @@
...
+  - Controller Manager:Creates manager Interface which implements events for managing ConfigmapManager LocationCache and podManager
+
+
+### Downstream Controller:Sync add/update/delete event to edge
```
Comment (anvithks) [inline docs/modules/cloud/controller.md]: When the document gets parsed for readthedocs all the headings with ## get added as a menu item in the table of contents and the sidebar menu.  
Use ## for any heading that  should be a part of the menu. Avoid long titles. 
If there is a longer title, keep a short name after ## and then a longer sub title under ###

File: docs/modules/cloud/controller.md [hunk 13]
```diff
@@ -0,0 +1,54 @@
...
+ 
+- Downstream controller:Watches K8S-Api-server and send updates to edge via cloudHub
+- Sync (pod,configmap,secret) add/update/delete event to edge via cloudHub
+- Creates Respective manager (pod,configmap,secret) for handling events by calling manager interface
+- Locates configmap and secret should be send to which node
+
+![Downstream Controller](./../../images/edgecontroller/DownstreamController.png)
+### Upstream Controller:sync watch and Update status of resource and events
```
Comment (anvithks) [inline docs/modules/cloud/controller.md]: Same as above regarding menu items

File: docs/modules/cloud/controller.md [hunk 14]
```diff
@@ -0,0 +1,54 @@
...
+
+- UpstreamController receives messages from edge and sync the updates to K8S-Api-server
+- Creates stop channel to dispatch and stop event to handle pods, configMaps ,node and secrets
```
Comment (anvithks) [inline docs/modules/cloud/controller.md]: nit: extra space after configMaps. no space after comma.

File: docs/modules/cloud/controller.md [hunk 15]
```diff
@@ -0,0 +1,54 @@
...
+- Creates message channel to update Nodestatus, Podstatus, Secret and configmap related events
+- Gets Podcondition information like Ready, initialized, Podscheduled and Unschedulable details
```
Comment (anvithks) [inline docs/modules/cloud/controller.md]: Consistent captalization or camel case. initialized -> Initialized

File: docs/modules/cloud/controller.md [hunk 16]
```diff
@@ -0,0 +1,54 @@
...
+- **Below is the information for PodCondition**
```
Comment (anvithks) [inline docs/modules/cloud/controller.md]: PodCondition or Podcondition. Consistent naming should be followed.

File: docs/modules/cloud/controller.md [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+   - **Ready**:PodReady means the pod is able to service requests and should be added to the load balancing pools for all matching services
+   - **PodScheduled**:It represents status of the scheduling process for this pod
+   - **Unschedulable**:It means scheduler cant schedule the pod right now,may be due to insufficient resources in the cluster
```
Comment (anvithks) [inline docs/modules/cloud/controller.md]: cant - > cannot.

Comment (anvithks) [inline docs/modules/cloud/controller.md]: nit: space after comma `now, may be`

File: docs/modules/cloud/controller.md [hunk 18]
```diff
@@ -0,0 +1,54 @@
...
+   - **Initialized**:It means that all Init containers in the pod have started sucessfully
+- Gives details about the Container status for the specific podsives 
+- Gives PodStatus information
+- **Below is the information for PodStatus**
+   - **PodPhase**:Current condition of the pod
+   - **Conditions**:Details indicating wht the pod is in this condition
```
Comment (anvithks) [inline docs/modules/cloud/controller.md]: Possible typo. `wht`

File: docs/modules/cloud/controller.md [hunk 19]
```diff
@@ -0,0 +1,54 @@
...
+   - **HostIP**:Ip address of the host to which pod is assigned
```
Comment (anvithks) [inline docs/modules/cloud/controller.md]: Ip -> IP

File: docs/modules/cloud/controller.md [hunk 20]
```diff
@@ -0,0 +1,54 @@
...
+   - **PodIp**:Ip address allocated to the Pod
```
Comment (anvithks) [inline docs/modules/cloud/controller.md]: Ip -> IP

File: docs/modules/cloud/controller.md [hunk 21]
```diff
@@ -0,0 +1,54 @@
...
+   - **QosClass**:Assigned to the pod based on resource requirement
+   
+   ![Upstream Controller](./../../images/edgecontroller/UpstreamController.png)
+   
+### Controller Manager:Creates manager Interface and implements ConfigmapManager LocationCache and podManager
```
Comment (anvithks) [inline docs/modules/cloud/controller.md]: Same as before regarding menu items.

File: docs/modules/cloud/controller.md [hunk 22]
```diff
@@ -0,0 +1,54 @@
...
+
+- Manager define the Interface of a manager,ConfigManager,Podmanager,secretmanager implements it
```
Comment (anvithks) [inline docs/modules/cloud/controller.md]: Please rephrase this statement.  
define -> defines  
nit: Spaces after commas.

File: docs/modules/cloud/controller.md [hunk 23]
```diff
@@ -0,0 +1,54 @@
...
+- Manages OnAdd, OnUpdate and OnDelete events which will be updated to the respective edge node from the K8s-Api-server
+- creates a eventManager(configMaps,pod,secrets) which will start a CommonResourceEventHandler, NewListWatch and a newShared Informer for each event to Sync(add/update/delete)event(pod, configmap, secret) to edge via cloudHub
```
Comment (anvithks) [inline docs/modules/cloud/controller.md]: creates -> Creates
Creates `an` eventManager..

File: docs/modules/cloud/controller.md (hunk 11 above)
Comment (srivatsav123) [reply docs/modules/cloud/controller.md]: ok

File: docs/modules/cloud/controller.md (hunk 14 above)
Comment (srivatsav123) [reply docs/modules/cloud/controller.md]: ok

**Pull request merged**